dataflow/src/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -133,6 +134,85 @@ public S getStoreBefore(Node node) {
     }
 
     /**
+     * @return the regular store immediately before a given {@link Block}
+     */
+    public S getStoreBefore(Block bb) {
+        TransferInput<A, S> transferInput = stores.get(bb);
+        AbstractAnalysis<A, S, ?> analysis = transferInput.analysis;
+        switch (analysis.getDirection()) {
+            case FORWARD:
+                {
+                    return transferInput.getRegularStore();
+                }
+            case BACKWARD:
+                {
+                    Node firstNode;
+                    switch (bb.getType()) {
+                        case REGULAR_BLOCK:
+                            firstNode = ((RegularBlock) bb).getContents().get(0);
+                            break;
+                        case EXCEPTION_BLOCK:
+                            firstNode = ((ExceptionBlock) bb).getNode();
+                            break;
+                        default:
+                            firstNode = null;
+                    }
+                    if (firstNode == null) {
+                        // this block doesn't contains any node, return store in transfer input
+                        return transferInput.getRegularStore();
+                    }
+                    return analysis.runAnalysisFor(firstNode, true, transferInput);
+                }
+            default:
+                {
+                    assert false;
```

Comment:
Use the ErrorReporter to always raise an error, not just when assertions are enabled. Provide a message.